anesthetic/samples.py

```diff
@@ -406,8 +411,14 @@ def __init__(self, *args, **kwargs):
             self.root = root
         else:
             self._beta = kwargs.pop('beta', 1.)
+            logL = kwargs.pop('logL', None)
             logL_birth = kwargs.pop('logL_birth', None)
-            super(NestedSamples, self).__init__(*args, **kwargs)
+            if logL is not None:
+                logL = numpy.where(logL <= logzero, -numpy.inf, logL)
+            if not isinstance(logL_birth, int) and logL_birth is not None:
+                logL_birth = numpy.where(logL_birth <= logzero,
+                                         -numpy.inf, logL_birth)
+            super(NestedSamples, self).__init__(logL=logL, *args, **kwargs)
```

Comment:
This code looks great. Is there any reason that this functionality for `logL` should be `NestedSamples` specific? It could easily be moved up into the parent `MCMCSamples` class. Is there a reason it shouldn't?

Reply:
It's been a while, so not 100% sure, but I think that should work just as well for `MCMCSamples`. Probably not needed there, since it doesn't go to as extreme values...?

Reply:
I guess that a metropolis hastings sampler would never accept 'logzero' values. However, First, these are `MCMCSamples`, not `MetropolisHastingsSamples`, and are therefore meant to be a bit more general than this. Second, I think this functionality could be useful for some analyses that like to increase the 'unphysical' cutoff artificially (such as capped likelihoods). I think it would be worth pushing this further up the hierarchy.

Reply:
Actually we had that discussion [above](https://github.com/williamjameshandley/anesthetic/pull/71#issuecomment-579385104):

>> I have only introduced logzero for NestedSamples so far. Should I also introduce it for MCMCSamples? The original problem is mostly connected to calculations of D and d...
> 
> I don't think logzero would affect any of the MCMCSamples functionality, so I'd keep it confined to NestedSamples

Do you still prefer moving it up in the hierarchy, now?

Reply:
I think I do prefer moving it up now, although if you disagree you can have the deciding vote between the three of us (Me, you and past me).

Reply:
moved up in hierarchy

Reply:
Do we need to add tests for using `logzero` with MCMCSamples?

Reply:
done anyhow